helix-term/src/ui/mod.rs

```diff
@@ -194,15 +195,11 @@ pub fn file_picker(root: PathBuf, config: &helix_view::editor::Config) -> FilePi
     // We want files along with their modification date for sorting
     let files = walk_builder.build().filter_map(|entry| {
         let entry = entry.ok()?;
-
         // This is faster than entry.path().is_dir() since it uses cached fs::Metadata fetched by ignore/walkdir
-        let is_dir = entry.file_type().map_or(false, |ft| ft.is_dir());
-        if is_dir {
-            // Will give a false positive if metadata cannot be read (eg. permission error)
-            None
-        } else {
-            Some(entry.into_path())
+        if !entry.file_type()?.is_file() {
+            return None;
         }
+        Some(entry.into_path())
```

Comment:
Might be easier to read as a positively expressed if
```suggestion
        if entry.file_type()?.is_file() {
            return Some(entry.into_path());
        }
        None
```